### PR TITLE
[flang][OpenMP] Add TODO messages for partially implemented atomic types

### PR DESCRIPTION
**Pull request by tblah**
There is ongoing work for fir.complex but this looks unlikely to land before the LLVM branch. Adding a TODO message gives cleaner output to users. Currently fir.complex leads to a compiler assertion failure.

Some uses of character types lead to invalid fir.convert type conversions, others make it far enough to hit the same assertion failure as for fir.complex.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
Will this warning hit for both OpenMP and OpenACC? If so, there might be a problem for OpenACC downstream users since they might have an implementation that works.

File: flang/lib/Lower/DirectivesCommon.h [hunk 1]
```diff
@@ -135,6 +136,22 @@ static inline void genOmpAtomicHintAndMemoryOrderClauses(
   }
 }
 
+template <typename AtomicListT>
+static void processAtomicTODO(mlir::Type elementType, mlir::Location loc) {
```
Comment (kiranchandramohan) [inline flang/lib/Lower/DirectivesCommon.h:140]:
```suggestion
static void processOmpAtomicTODO(mlir::Type elementType, mlir::Location loc) {
```

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

**Review by pawosm-arm — APPROVED** [review]

**Review by NimishMishra — APPROVED** [review]
LGTM

**Pull request merged**